clients/unshaded/src/main/java/tachyon/client/file/BaseFileSystem.java

```diff
@@ -0,0 +1,276 @@
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      return masterClient.loadMetadata(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public void mount(TachyonURI src, TachyonURI dst) throws IOException, TachyonException {
+    mount(src, dst, MountOptions.defaults());
+  }
+
+  @Override
+  public void mount(TachyonURI src, TachyonURI dst, MountOptions options)
+      throws IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.mount(src, dst);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public FileInStream openFile(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    return openFile(path, OpenFileOptions.defaults());
+  }
+
+  @Override
+  public FileInStream openFile(TachyonURI path, OpenFileOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    URIStatus status = getStatus(path);
+    return new FileInStream(status.getInfo(), options.toInStreamOptions());
+  }
+
+  @Override
+  public void rename(TachyonURI src, TachyonURI dst)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    rename(src, dst, RenameOptions.defaults());
+  }
+
+  @Override
+  public void rename(TachyonURI src, TachyonURI dst, RenameOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.rename(src, dst);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public void setAttribute(TachyonURI path)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    setAttribute(path, SetAttributeOptions.defaults());
+  }
+
+  @Override
+  public void setAttribute(TachyonURI path, SetAttributeOptions options)
+      throws FileDoesNotExistException, IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.setAttribute(path, options);
+    } finally {
+      FileSystemContext.INSTANCE.releaseMasterClient(masterClient);
+    }
+  }
+
+  @Override
+  public void unmount(TachyonURI path) throws IOException, TachyonException {
+    unmount(path, UnmountOptions.defaults());
+  }
+
+  @Override
+  public void unmount(TachyonURI path, UnmountOptions options)
+      throws IOException, TachyonException {
+    FileSystemMasterClient masterClient = FileSystemContext.INSTANCE.acquireMasterClient();
+    try {
+      masterClient.unmount(path);
```

Comment:
currently, the thrift interface does not support unmount options and the client-side UnmountOptions struct has no options